src/inventory.js

```diff
@@ -0,0 +1,205 @@
+class Inventory extends Phaser.Scene {
+
+    constructor() {
+        super({key: "Inventory"});
+        this.var = 10;
+    }
+
+    init(data) {
+        this.player = data.player;
+    }
+
+    preload() {
+
+        eqPreload(this);
+
+        this.load.image("inventory_background",
+          "../assets/tiles/inventory.png");
+    }
+
+    create() {
+
+        this.drawBackground();
+        this.drawItems();
+        prepareKeyDownListenersInv(this);
+    }
+
+    drawBackground(){
+      this.add.image(370,300, 'inventory_background');
+    }
+
+    drawItems() {
+
+        showInventoryTitle(this);
+        group.inventory = this.physics.add.group();
+
+        for( var i =0 ; i <   this.player.eq.length ; i++ )
+        {
+            var itemName = this.player.eq[i];
+            var item = ITEMS_KINDS.find(function(element) {
+                return element.name == itemName;
+            });
+
+            group.inventory.create(200,215+i*20, itemName ,0).setInteractive();
+
+            this.text = this.add.text(220, 200+i*20 ,itemName , {
+                fill: '#000',
+                fontSize: '20px',
+                fontFamily: 'Comic Sans MS'
+            });
+
+            this.text = this.add.text(350, 200+i*20, item.describ , {
+                fill: '#000',
+                fontSize: '20px',
+                fontFamily: 'Comic Sans MS'
+            });
+            this.add.sprite(175,215+i*20,EqInfo.SPECIAL_MARKER().name)
+        }
+
+        for( var i =0 ; i < this.player.items.length ; i++ )
+        {
+            var itemName = this.player.items[i];
+            var item = ITEMS_KINDS.find(function(element) {
+                return element.name == itemName;
+            });
+
+            group.inventory.create(200,215+(i+this.player.eq.length)*20, itemName ,0).setInteractive();
+
+            this.text = this.add.text(220, 200+(i+this.player.eq.length)*20 ,itemName , {
+                fill: '#000',
+                fontSize: '20px',
+                fontFamily: 'Comic Sans MS'
+            });
+
+            this.text = this.add.text(350, 200+(i+this.player.eq.length)*20, item.describ , {
+                fill: '#000',
+                fontSize: '20px',
+                fontFamily: 'Comic Sans MS'
+            });
+        }
+
+        this.input.on('gameobjectup', this.clickHandler, this);
+    }
+
+    clickHandler (pointer, item) {
+
+        var itemName = item.texture.key;
+        console.log("clicked: "+itemName)
+
+        if (itemName == "POTION") {
```

Comment:
I know that it can be implemented more generic. But in our case it  should be efficient